carbon-design-system/carbon-addons-iot-react
# 2439 sass modules

Closes #2439 
Closes #2440 
Closes #2441 
Closes #2442 
Closes #2443 
Closes #2444 

## Depends on #3055 

**Summary**

- Writes all the scss to use `@use` instead of `@import`. This brings in the variables within scope, so all variables are now referenced using their scope. This upgrades stylelint, sass, and adds all the new carbon packages necessary. `carbon-components` is being replaced by `@carbon/styles`. 

**Change List (commits, features, bugs, etc)**

- Upgrades all scss files to use `@use` instead of `@import`
- - updated `carbon-components` to `@carbon/styles` where necessary
- upgrades stylelint, postcss
- wrote a custom loader/importer to get around `rollup-plugin-postcss` issues with new sass versions

**Acceptance Test (how to verify the PR)**

- check there are no usages of `@import` in our styles

**Regression Test (how to make sure this PR doesn't break old functionality)**

- check for visual regressions in storybook



**Things to look for during review**
